packages/hardhat-viem/src/internal/errors.ts

```diff
@@ -57,6 +57,16 @@ const contractB = await hre.viem.getContractAt("B", address, { walletClient });`
   }
 }
 
+export class InvalidConfirmationsError extends HardhatViemError {
+  constructor(confirmations: number) {
+    super(
+      confirmations < 0
```

Comment:
Can the variable confirmations be > 0 ?

Reply:
What do you mean? Confirmations are expected to be higher than 0. If it is below 0, we want to throw an error. I don't expect this error to be invoked with confirmations > 0 (see [this](https://github.com/NomicFoundation/hardhat/pull/4426/files#diff-bd2b6167ff354b394e83027a2bb3d5f28b236d392dc745de9c637e0aa72bf5e7R68)). Do you think we need to double check?

Reply:
I was also confused by this code. I think I would split this into two errors: one for the case where someone uses 0 and another one for a negative number. The latter doesn't even need to be a custom error, it can be the generic `HardhatViemError` with a proper message.